test/e2e/cluster_upgrade.go

```diff
@@ -0,0 +1,136 @@
+/*
+Copyright 2015 The Kubernetes Authors All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package e2e
+
+import (
+	"fmt"
+	"net/http"
+	"os"
+	"os/exec"
+	"path"
+	"sync"
+	"time"
+
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/api"
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/fields"
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/labels"
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/util"
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/util/wait"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+)
+
+var _ = Describe("Skipped", func() {
+	Describe("Cluster upgrade", func() {
+		name := "baz"
+		framework := Framework{BaseName: "cluster-upgrade"}
+		var webserver *WebserverTest
+
+		BeforeEach(func() {
+			framework.beforeEach()
+			webserver = NewWebserverTest(framework.Client, framework.Namespace.Name, name)
+			webserver.CreateWebserverPod()
+			svc := webserver.BuildServiceSpec()
+			svc.Spec.Type = api.ServiceTypeLoadBalancer
+			webserver.CreateService(svc)
+		})
+
+		AfterEach(func() {
+			framework.afterEach()
+			webserver.Cleanup()
+		})
+
+		It("by kube-push of master should maintain responsive services", func() {
+			testClusterUpgrade(framework, name, func() {
+				runUpgradeScript("hack/e2e-internal/e2e-push.sh", "-m")
+			})
+		})
+
+		It("by gce-upgrade of master should maintain responsive services", func() {
+			if !providerIs("gce") {
+				By(fmt.Sprintf("Skippingt test, which is not implemented for %s", testContext.Provider))
+				return
+			}
+			testClusterUpgrade(framework, name, func() {
+				runUpgradeScript("hack/e2e-internal/e2e-upgrade.sh", "-M", "-l")
+			})
+		})
+	})
+})
+
+func testClusterUpgrade(framework Framework, svcName string, upgrade func()) {
+	result, err := waitForLoadBalancerIngress(framework.Client, svcName, framework.Namespace.Name)
+	Expect(err).NotTo(HaveOccurred())
+	if len(result.Status.LoadBalancer.Ingress) != 1 {
+		Failf("got unexpected number (%v) of ingress points for externally load balanced service, only expecting one: %v", result.Status.LoadBalancer.Ingress, result)
```

Comment:
nits:
- do you want the first var to be `%d` and `len(result.Status.LoadBalancer.Ingress)`?
- I'd use "1" rather than "one" to be consistent
- I'm imagining from just reading the output this could be a bit more clear about what's being printed (are we expecting  `result.Status.LoadBalancer.Ingress` or `result`?). If you do want to show both, I'd make it clear which you're showing.